tests/README.md

```diff
@@ -27,64 +27,65 @@ accessible via ssh on the libvirt domain and have Ansible installed on the host
 
 *NOTE*: Create snapshot after all these steps, to manually revert in case the
 test suite breaks something and fails to revert. Do not use snapshot names
-starting with `ssg\_`
+starting with `ssg_`
 
-## Two modes of operation
-SSG Test Suite currently supports two ways of DataStream evaluation. Simpler
-one to use is `profile`, which scans and remedies target domain based on
-particular profile.
+## Executing the test suite
 
-Second, more complex, is `rule`, which runs set of validation scenarios
-for particular subset of rules. Each rule has its own scenarios, and each
-scenario is run separately to eliminate need for cleanup.
+### Common options
 
-## Running test suite in profile mode
-Example of evaluation of default profile (common):
+- `--hypervisor`: Typically, you will use the `qemu:///system` value.
+- `--domain`: `libvirt` domain, which is basically name of the virtual machine.
+- `--datastream`: Path to the datastream that you want to use for scanning.
+  It will be transferred to the scanned VM via SSH.
+- `--xccdf-id`: Also known as `Ref-ID`, it is the identifier of benchmark within the datastream.
+  It can be obtained by running `oscap info` over the datastream XML and searching for `Ref-ID` strings.
+- `--help`: This will get you the invocation help.
+
+### Profile-based testing
+In this operation mode, you specify the `profile` command and you supply the profile ID as a positional argument.
+The test suite then runs scans over the target domain and remediates it based on particular profile.
+
+An example invocation may look like this:
 
 ```
 ./test_suite.py profile --hypervisor qemu:///system --domain ssg-test-suite-centos --datastream ssg-centos7-ds.xml --xccdf-id xccdf-id profile-id
 ```
 
-where the domain (in this case `ssg-test-suite-centos`), is name of the virtual machine,
-datastream is a datastream file in the local filesystem,
-the `xccdf-id` can be obtained by running `oscap info` over the datastream XML (look for `Ref-ID`),
-and
-profile-id is not matched by the suffix, so specify it literally (use `oscap info --profiles` to see available profiles).
+Note that the `profile-id` is not matched by the suffix (as opposed to how the `oscap` tool works), so specify it literally (use `oscap info --profiles` to see available profiles).
 
-For further options, see
+### Rule-based testing
+In this mode, you specify the `rule` command and you supply part of the rule title as a positional argument.
+Unlike the profile mode, here each rule from the matching rule set is addressed independently, one-by-one.
 
-```
-./test_suite.py profile --help
-```
+Rule-based testing enables to perform two kinds of tests:
 
+- Check tests: The system is set up into a compliant, or a non-compliant state.
 
-## Running test suite in rule mode
-Example of evaluation of rule ```rule_sysctl_net_ipv4_conf_default_secure_redirects```:
+  Typically, the compliant state is different from the default or post-remediation state.
```

Comment:
Isn't the post-remediation state after the run of remediation? Shoudln't the system be compliant then?

Reply:
What I wanted to say was that the compliant state induced by the script is typically different than compliant state induced by a remediation. Do you think that it is worth to clarify this further?

Reply:
I see, since an OVAL definition can return `pass` for multiple states, and while a remediation script only brings the system into a compliant state, it is good to test these other states in which result is `pass`.

I'm just not sure that it is that common for OVAL definitions have multiple states in which the system pass. Anyway, I think that clarification of why a compliant state can be different of a post-remediation state is enough.